main.py

```diff
@@ -163,128 +158,30 @@ def main():
+            # model = linear.train_1vsrest(y, x)
+            pass
         else:
-            word_dict = data_utils.load_or_build_text_dict(
-                dataset=datasets['train'],
-                vocab_file=config.vocab_file,
-                min_vocab_freq=config.min_vocab_freq,
-                embed_file=config.embed_file,
-                embed_cache_dir=config.embed_cache_dir,
-                silent=config.silent,
-                normalize_embed=config.normalize_embed
-            )
-            classes = data_utils.load_or_build_label(datasets, config.label_file, config.silent)
-            network = getattr(networks, config.model_name)(
-                embed_vecs=word_dict.vectors,
-                num_classes=len(classes),
-                **dict(config.network_config)
-            )
-            if config.init_weight is not None:
-                init_weight = networks.get_init_weight_func(
-                    init_weight=config.init_weight)
-                network.apply(init_weight)
-
-            model = Model(
-                classes=classes,
-                word_dict=word_dict,
-                network=network,
-                log_path=log_path,
-                **dict(config)
-            )
-
-        # Set up dataset loader
-        train_loader = data_utils.get_dataset_loader(
-            data=datasets['train'],
-            word_dict=model.word_dict,
-            classes=model.classes,
-            device=device,
-            max_seq_length=config.max_seq_length,
-            batch_size=config.batch_size,
-            shuffle=config.shuffle,
-            data_workers=config.data_workers
-        )
-        val_loader = data_utils.get_dataset_loader(
-            data=datasets['val'],
-            word_dict=model.word_dict,
-            classes=model.classes,
-            device=device,
-            max_seq_length=config.max_seq_length,
-            batch_size=config.eval_batch_size,
-            data_workers=config.data_workers
-        )
-
-        # Start training
-        trainer.fit(model, train_loader, val_loader)
-        logging.info(f'Loading best model from `{checkpoint_callback.best_model_path}`...')
-        model = Model.load_from_checkpoint(checkpoint_callback.best_model_path)
-
-    if 'test' in datasets:
-        test_loader = data_utils.get_dataset_loader(
-            data=datasets['test'],
-            word_dict=model.word_dict,
-            classes=model.classes,
-            device=device,
-            max_seq_length=config.max_seq_length,
-            batch_size=config.eval_batch_size,
-            data_workers=config.data_workers
-        )
-        trainer.test(model, test_dataloaders=test_loader)
-        if config.save_k_predictions > 0:
-            if not config.predict_out_path:
-                config.predict_out_path = os.path.join(checkpoint_dir, 'predictions.txt')
-            save_predictions(trainer, model, test_loader, config.predict_out_path)
```

Comment:
284-287 not handled in the new nn training file?? Not nobody calls save_predictions?